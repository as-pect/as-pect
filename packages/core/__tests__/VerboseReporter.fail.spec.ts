import { VerboseReporterWrapper } from "./setup/VerboseReporterWrapper.js";
import { TestContext } from "../src/index.js";
import { promises as fs } from "fs";
import { instantiate } from "@assemblyscript/loader";

test("snapshots", async () => {
  const binary = await fs.readFile("./assembly/jest-reporter-fail.wasm");
  const reporter = new VerboseReporterWrapper();
  const ctx = new TestContext({
    binary,
    reporter,
    fileName: "./assembly/jest-reporter-fail.ts",
  });
  ctx.run(await instantiate(binary, ctx.createImports()));
  for (const [name, values] of reporter.snapshots.entries()) {
    for (const value of values) {
      expect(value).toMatchSnapshot(name);
    }
  }
});
