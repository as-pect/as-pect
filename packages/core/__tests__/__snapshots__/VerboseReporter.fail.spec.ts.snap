// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots: onFinish ./assembly/jest-reporter-fail.ts 1`] = `
"
   [Error]: ReflectedValue Cannot log ReflectedValue of id -1. Index out of bounds.
   [Stack]: Has Stack Trace
    [File]: ./assembly/jest-reporter-fail.ts
  [Groups]: 2 pass, 2 total
  [Result]: ✖ FAIL
[Snapshot]: 0 total, 0 added, 0 removed, 0 different
 [Summary]: 4 pass,  5 fail, 9 total
    [Time]: 0ms

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

"
`;

exports[`snapshots: onGroupEnd  1`] = `""`;

exports[`snapshots: onGroupEnd examples 1`] = `
"    [Todo]: example todo
     [Log]: \\"testing in a group\\"
   [Stack]: Has Stack Trace

"
`;

exports[`snapshots: onGroupStart examples 1`] = `
"[Describe]: examples

"
`;

exports[`snapshots: onTestFinish examples example 1 1`] = `
" [Success]: ✔ example 1 RTrace: +7
"
`;

exports[`snapshots: onTestFinish examples example 1 2`] = `
" [Success]: ✔ example 1 RTrace: +7
"
`;

exports[`snapshots: onTestFinish examples log value 1`] = `
" [Success]: ✔ log value RTrace: +3
     [Log]: \\"testing in a test\\"
   [Stack]: Has Stack Trace
"
`;

exports[`snapshots: onTestFinish examples push an error to the error array 1`] = `
"    [Fail]: ✖ push an error to the error array
   [Stack]: No Stack Trace
"
`;

exports[`snapshots: onTestFinish examples something fails 1`] = `
"    [Fail]: ✖ something fails
  [Actual]: -1
[Expected]: 42
 [Message]: Here is an example failure message.
   [Stack]: Has Stack Trace
"
`;

exports[`snapshots: onTestFinish examples something negated fails 1`] = `
"    [Fail]: ✖ something negated fails
  [Actual]: 42
[Expected]: Not 42
 [Message]: 42 is 42
   [Stack]: Has Stack Trace
     [Log]: \\"another string\\"
   [Stack]: Has Stack Trace
"
`;

exports[`snapshots: onTestFinish examples something using toBe fails with a reference that is similar 1`] = `
"    [Fail]: ✖ something using toBe fails with a reference that is similar
  [Actual]: Vec3 {
      x: 1.0 as f32,
      y: 2.0 as f32,
      z: 3.0 as f32
  }
[Expected]: \\"Serializes to same value.\\"
 [Message]: not the same pointer
   [Stack]: Has Stack Trace
"
`;

exports[`snapshots: onTestFinish examples this should throw but it does not 1`] = `
"    [Fail]: ✖ this should throw but it does not
 [Message]: it should throw
   [Stack]: Has Stack Trace
"
`;

exports[`snapshots: onTestFinish examples when two things should equal each other 1`] = `
"  [Throws]: ✔ when two things should equal each other RTrace: +15
"
`;

exports[`snapshots: onTestStart examples example 1 1`] = `""`;

exports[`snapshots: onTestStart examples example 1 2`] = `""`;

exports[`snapshots: onTestStart examples log value 1`] = `""`;

exports[`snapshots: onTestStart examples push an error to the error array 1`] = `""`;

exports[`snapshots: onTestStart examples something fails 1`] = `""`;

exports[`snapshots: onTestStart examples something negated fails 1`] = `""`;

exports[`snapshots: onTestStart examples something using toBe fails with a reference that is similar 1`] = `""`;

exports[`snapshots: onTestStart examples this should throw but it does not 1`] = `""`;

exports[`snapshots: onTestStart examples when two things should equal each other 1`] = `""`;
